bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSBridgeHandler.java

```diff
@@ -0,0 +1,362 @@
+            client.sendCommand(new ServerListeningCommand());
+        }
+    }
+
+    public void sendCommand(BaseCommand command) {
+        if (getThing().getStatus().equals(ThingStatus.ONLINE)) {
+            client.sendCommand(command);
+        }
+    }
+
+    public @Nullable Node requestNodeDetails(int nodeId) {
+        Node node = lastNodeStates.get(nodeId);
+        logger.debug("Node {}. Details requested, provided: {}", nodeId, node != null);
+        return node;
+    }
+
+    @Override
+    public void registerNodeListener(ZwaveNodeListener nodeListener) {
+        final Integer id = nodeListener.getId();
+        if (nodeListeners.put(id, nodeListener) != null) {
+            logger.debug("Node {}. Registering listener", id);
+        }
+    }
+
+    @Override
+    public boolean unregisterNodeListener(ZwaveNodeListener nodeListener) {
+        logger.debug("Node {}. Unregistering listener", nodeListener.getId());
+        return nodeListeners.remove(nodeListener.getId()) != null;
+    }
+
+    public boolean registerDiscoveryListener(NodeDiscoveryService listener) {
+        logger.debug("Registering Z-Wave discovery listener");
+        if (discoveryService == null) {
+            discoveryService = listener;
+            getFullState();
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean unregisterDiscoveryListener() {
+        logger.debug("Unregistering Z-Wave discovery listener");
+        if (discoveryService != null) {
+            discoveryService = null;
+            return true;
+        }
+
+        return false;
+    }
+
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(NodeDiscoveryService.class, ZwaveJSActions.class);
+    }
+
+    @Override
+    public void dispose() {
+        stopInitialConnectionJob();
+        client.stop();
+        super.dispose();
+    }
+
+    @Override
+    public void onConnectionError(String message) {
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, message);
+    }
+
+    public void startInclusion() {
+        sendCommand(new ControllerInclusionCommand(false));
+    }
+
+    public void stopInclusion() {
+        sendCommand(new ControllerInclusionCommand(true));
+    }
+
+    public void startExclusion() {
+        sendCommand(new ControllerExclusionCommand(false));
+    }
+
+    public void stopExclusion() {
+        sendCommand(new ControllerExclusionCommand(true));
+    }
+
+    public void sendMulticastCommand(String nodeIDs, Integer commandClass, Integer endpoint, String property,
+            String value) {
+        sendCommand(new MulticastSetValueCommand(parseNodeIDs(nodeIDs), commandClass, endpoint, property,
+                convertValueType(value)));
+    }
+
+    private static int[] parseNodeIDs(String nodeIDs) {
+        return Arrays.stream(nodeIDs.split(",")).map(String::trim).mapToInt(Integer::parseInt).toArray();
```

Comment:
There is a risk of exception with parseInt.
No need to catch the exception ?

Reply:
Improved.